tests/x11/vnc_two_passwords.pm

```diff
@@ -85,7 +85,7 @@ sub generate_vnc_events {
     mouse_click;
 
     send_key 'alt-f4';
-    wait_serial 'vncviewer-finished';
+    wait_serial('vncviewer-finished') || die 'vncviewer not finished';
```

Comment:
It's wrong 😕 you're hiding a test issue. The problem is the generation of the vnc events, that is causing a race condition. And the situation is very similar with vino test module

Reply:
A fail is not hiding the issue. Green is hiding the issue.

Reply:
Well, to put it better: You're failing a test module, because you're covering for an issue with the test itself, and not the product.

Reply:
The current state is misleading the real issue: https://openqa.suse.de/tests/4186174#step/vnc_two_passwords/24

It tells that the wc command fails. But, the test should fails at wait_serial, because the problem is the command that wait_serial is waiting for.

Reply:
> Well, to put it better: You're failing a test module, because you're covering for an issue with the test itself, and not the product.

The test is failing if wait_serial doesn't find the string.

Example:

```perl
script_run('command a');
assert_script_run('command b');
```

The test module should not continue if `command a` fails. With that code, when `command a` fails, `command b` fails and test fails with message `command b timed out`

This PR is the equivalent to change `script_run` to `assert_script_run`.

Example:

```perl
assert_script_run('command a');
assert_script_run('command b');
```

With that change, when `command a` fails, the test will failed with message `command a timed our`. This is not hiding not covering any bug and is reporting the right command to look at to investigate the failure.

Is you concern about the `die` message? I can change it to `vncviewer ... timed out`.

Reply:
> > Well, to put it better: You're failing a test module, because you're covering for an issue with the test itself, and not the product.
> 
> The test is failing if `wait_serial` doesn't find the string.

Yes, and it's doing it's job already: https://openqa.suse.de/tests/4037949#step/vnc_two_passwords/19

> Is you concern about the `die` message? I can change it to `vncviewer ... timed out`.

I just can't really see the benefit of the PR other than improving message reporting. And the ticket actually mentions the source of the problem: something inside `generate_vnc_events` is either causing a window to be out of focus _OR_ the more likely: the `wait_serial` needs more time to actually send the info (maybe the trap doesn't always gets properly called or something, or the wrong window is closed)

Reply:
> I just can't really see the benefit of the PR other than improving message reporting.

Please, remember how many time was invested (wasted?) in failures by QSF-U team because the wrong message was reported.
Also, how can you be against a better reporting?


> And the ticket actually mentions the source of the problem: something inside `generate_vnc_events` is either causing a window to be out of focus _OR_ the more likely: the `wait_serial` needs more time to actually send the info (maybe the trap doesn't always gets properly called or something, or the wrong window is closed)

Is your complaint because this PR doesn't solve the ticket? Of course not, I have three commits, but I wanted to avoid endless discussions and push one commit at a time to reduce the "attack surface".
See: #10040